### ISSUE TITLE
[v2] Building with `--prefix-paths` is broken.

### ISSUE THREAD

**Issue by sktt** [gatsbyjs/gatsby#7163]
Building with `--prefix-paths` will (1) generate a build without any static markup and (2) will not prefix `<Link />` components' `to` prop.

Steps to reproduce (1):
- `gatsby new bug https://github.com/gatsbyjs/gatsby-starter-default\#v2`
- add `pathPrefix: 'bug',` to `gatsby-config.js`
- `gatsby build --prefix-paths`
- `cat public/index.html | grep Welcome || echo 'Broken!'`

Expected result: 
```
[..]<h1>Hi people</h1><p>Welcome to your new Gatsby site.</p><p>Now go build something great.</p>[..]
```
Actual result:
```
Broken!
```

Next to observe (2): 
- `mv public bug`
- `python -mSimpleHTTPServer`
- http://localhost:8000/bug
- Observe the link to /page-2

Expected:
Should be `/bug/page-2`

Actual:
`/page-2`

**Comment by KyleAMathews** [comment]
Thanks for the bug report w/ nice reproduction steps! PR with fix is up @ #7176

**Comment by m-allanson** [comment]
#7176 was a partial fix, reopening this.

**Comment by m-allanson** [comment]
This should be fixed in `gatsby@2.0.0-beta.93`

**Comment by izorg** [comment]
@m-allanson looks like now during build `location.pathname` has additional `/` if no `pathPrefix`. For index page it's `//`
